src/rewriter.coffee

```diff
@@ -655,7 +655,7 @@ exports.Rewriter = class Rewriter
   # primitive string and separately passing any expected token data properties
   exposeTokenDataToGrammar: ->
     @scanTokens (token, i) ->
-      if token.data and Object.keys(token.data).length or token[0] is 'JS' and token.generated
+      if token.data and Object.keys(token.data).length or token[0] in ['JS', 'CALL_START'] and token.generated
```

Comment:
From my understanding, `generated` is a preexisting pattern that sets `generated` directly on `token` rather than using the token data mechanism. So this wasn't catching it

I guess one option would be to update all current uses of `generated` to be set as token data? Not sure if that's the type of refactor you'd rather treat separately

Reply:
Maybe rather than looking for token tag, should we look for `.generated`?

```coffee
if token.generated or (token.data and Object.keys(token.data).length isnt 0)
```

I guess the question is, what’s the harm in applying this for all generated tokens rather than only generated `JS` and `CALL_START` ones. Are those two special because they’re generated in the rewriter? Are they the only token types generated in the rewriter, and if not, should we include the other types (or if not, why not)?

Just wanting to make sure we understand what’s going on here. A comment explaining it might be good to add.